notebooks/problem-sets/quantum_enigma_004.ipynb

```diff
@@ -0,0 +1,334 @@
+    "tags": []
+   },
+   "source": [
+    "# Enigma 004 : The Monty Hall Problem"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "![University of Sherbrooke, Institute of Quantique](images/IQ_Logo.png)\n",
+    "\n",
+    "## Overview\n",
+    "\n",
+    "Watch the following video before attempting this problem set\n",
+    "\n",
+    "<div class=\"youtube-wrapper\">\n",
+    "    <iframe width=\"560\" height=\"315\" src=\"https://www.youtube.com/embed/Hd9KhRts1uw\" title=\"YouTube video player\" frameborder=\"0\" allow=\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture\" allowfullscreen></iframe>\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Problem 1\n",
+    "\n",
+    "Write a circuit that would only use 3 qubits instead of 4 for Enigma 004 The Monty Hall Problem.\n",
+    "\n",
+    "<!-- ::: q-block.reminder -->\n",
+    "\n",
+    "### Hints\n",
+    "\n",
+    "<details>\n",
+    "    <summary>Hint 1</summary>\n",
+    "    Use only two qubits to hide the diamond.\n",
+    "</details>\n",
+    "\n",
+    "<details>\n",
+    "    <summary>Hint 2</summary>\n",
+    "    Find a way to obtain $\\frac{1}{3}$ probability of measuring `00`, `01`, and `10` (the binary equivalent of 0, 1, and 2).\n",
+    "</details>\n",
+    "<!-- ::: -->\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "goals": [
+     {
+      "id": "enigma-004-quiz-1"
+     }
+    ],
+    "grader_answer": "problem_qc",
+    "grader_id": "quantum_enigmas/enigma_004_problem_1"
+   },
+   "outputs": [],
+   "source": [
+    "from qiskit import QuantumCircuit, ClassicalRegister, QuantumRegister\n",
+    "import matplotlib\n",
+    "from qiskit.circuit.library import MCXGate\n",
+    "from math import pi\n",
+    "\n",
+    "problem_qc = QuantumCircuit(3, 3)\n",
+    "\n",
+    "# Start your work here.\n",
+    "# We've provided the circuit that is shown above\n",
+    "# Your circuit MUST be named problem_qc\n",
+    "\n",
+    "\n",
+    "problem_qc.draw(\"mpl\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Problem 2\n",
+    "\n",
+    "Write a circuit that would randomly determine the chest you choose at the start (using three qubits for hiding the diamond as in the enigma).\n",
+    "\n",
+    "<!-- ::: q-block.reminder -->\n",
+    "\n",
+    "### Hints\n",
+    "\n",
+    "<details>\n",
+    "    <summary>Hint 1</summary>\n",
+    "    You can use $q_3$, $q_4$, and $q_5$ to determine the safe you put your hand on at first and $q_6$, $q_7$, and $q_8$ to determine which safe will be opened (safe 0 is linked to $q_0$, $q_3$, and $q_6$; safe 1 is linked to $q_1$, $q_4$, and $q_7$; safe 2 is linked to $q_2$, $q_5$, and $q_8$).\n",
+    "</details>\n",
+    "\n",
+    "<details>\n",
+    "    <summary>Hint 2</summary>\n",
```

Comment:
If we cannot put the code in Hint 2, we should then put Hint 2 text as an explanation just above the code (and not in a second hint) or it could be an image added to Hint 2